automation/docker/run-tests.sh

```diff
@@ -1,73 +1,79 @@
 #!/bin/bash
 
+# Get script location, via https://stackoverflow.com/a/12197227
+pushd . > /dev/null
+WORKING_DIR="${BASH_SOURCE[0]}";
+  while([ -h "${WORKING_DIR}" ]) do
+    cd "`dirname "${WORKING_DIR}"`"
+    WORKING_DIR="$(readlink "`basename "${WORKING_DIR}"`")";
+  done
+cd "`dirname "${WORKING_DIR}"`" > /dev/null
+WORKING_DIR="`pwd`";
+popd  > /dev/null
+
+if [ -z ${1+x} ]; then
+  printf "Must specify where Firecloud is running: 'fiab', 'local', 'alpha', 'prod', or ip."
+  exit 1
+fi
+
+# Defaults
+ENV=dev
+NUM_NODES=2
+TEST_ENTRYPOINT="testOnly -- -l ProdTest"
+TEST_CONTAINER="automation-$(cat /dev/urandom | env LC_CTYPE=C tr -dc 'a-z0-9' | fold -w 8 | head -n 1)"
+
 # Parameters
-NUM_NODES="${1:-4}"  # default to 4
-ENV="${2:-dev}"  # default to dev
-export ENV=$ENV
-HUB_COMPOSE=hub-compose-fiab.yml
-
-#if test runs against a remote FIAB on a GCE node, put IP in param 3
-#if test runs against a local FIAB on a Docker, put "local" in param 3
-#else leave blank (test runs against a non FIAB host, such as real dev)
-DOCKERHOST="127.0.0.1"
-DOCKERHOST=${3:-$DOCKERHOST}
-if [ $DOCKERHOST = "alpha" -o $DOCKERHOST = "prod" ];
-  then
-    DOCKERHOST=
-    HUB_COMPOSE=hub-compose.yml
+FC_INSTANCE=${1}
+ENV=${2:-$ENV}
+VAULT_TOKEN=${3:-$(cat ~/.vault-token)}
+
+export FC_INSTANCE WORKING_DIR ENV
+
+if [ "$ENV" = "prod" ]; then
+  TEST_ENTRYPOINT="testOnly -- -n ProdTest"
 fi
-export DOCKERHOST=$DOCKERHOST
-TEST_CONTAINER=${4:-automation}
-VAULT_TOKEN=$5
-WORKING_DIR=${6:-$PWD}
-export WORKING_DIR=$WORKING_DIR
-
-if [ -z $VAULT_TOKEN ]; then
-    VAULT_TOKEN=$(cat ~/.vault-token)
+
+if [ "$FC_INSTANCE" = "local" ]; then
+  FC_INSTANCE="172.19.0.1"
```

Comment:
where did this ip address come from?

Reply:
It's reliably the address for the real localhost inside docker networks. See the firecloud-ui rsync script.

Reply:
I would document this something like as follows:
```
HOST_ADDRESS="172.19.0.1" # maybe link to doc here
...
if ...
  FC_INSTANCE="$HOST_ADDRESS"
```